### PR TITLE
Teacher Dashboard/ update Manage Classes UI

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
update Manage Classes UI of the teacher dashboard

## WHY
we want to bring better design parity to this page

## HOW
various React/CSS tweals

### Screenshots
(If applicable. Also, please censor any sensitive data)
Screen Shot 2024-10-25 at 2 57 02 PM
Screen Shot 2024-10-25 at 2 57 28 PM
Screen Shot 2024-10-25 at 2 57 13 PM

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/Update-UI-My-Classes-3684a924e5524a2a8a6ec88b81297f58?pvs=4

### What have you done to QA this feature?
(Provide enough detail that a reviewer could assess whether additional QA should be done. For larger projects, additionally use the Engineer Feature Testing Notion template. Review Guidelines if needed: https://www.notion.so/quill/Github-PR-QA-Guidelines-49e99fc965654ceeb8c6249bd9d181d7)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  (The answer should mostly be 'YES'. If you answer 'NO', please justify.)
Have you deployed to Staging? | (Possible answers: YES, Not yet - deploying now!, NO - non-app change, NO - tiny change)
Self-Review: Have you done an initial self-review of the code below on Github? |

### CODE REVIEW

File: services/QuillLMS/client/app/bundles/Shared/components/shared/dataTable.tsx [hunk 1]
```diff
@@ -264,7 +264,11 @@ export class DataTable extends React.Component<DataTableProps, DataTableState> {
   }
 
   renderActions(row) {
-    if (!row.actions) { return }
+    if (!row.actions) {
+      return(
+        <td className="data-table-row-section actions-section" key={row.id}><p className="no-action">--</p></td>
+      )
+    }
```
Comment (eadams17) [inline services/QuillLMS/client/app/bundles/Shared/components/shared/dataTable.tsx:272]: confirmed with Jack that we want this universal change to display dashes when no action is present

Comment (brendanshean) [reply services/QuillLMS/client/app/bundles/Shared/components/shared/dataTable.tsx:272]: I would extract out a constant `NO_ACTION='---'`

File: services/QuillLMS/client/app/bundles/Teacher/components/classrooms/active_classrooms.tsx [hunk 2]
```diff
@@ -45,7 +45,7 @@ const bookEmptySrc = `${process.env.CDN_URL}/images/illustrations/book-empty.svg
 const canvasIconSrc = `${process.env.CDN_URL}/images/icons/canvas.svg`
 const cleverIconSrc = `${process.env.CDN_URL}/images/icons/clever.svg`
 const googleClassroomIconSrc = `${process.env.CDN_URL}/images/icons/google-classroom.svg`
-const reorderSrc = `${process.env.CDN_URL}/images/icons/reorder.svg`
+const reorderSrc = `${process.env.CDN_URL}/images/icons/s/reorder.svg`
```
Comment (brendanshean) [inline services/QuillLMS/client/app/bundles/Teacher/components/classrooms/active_classrooms.tsx]: `/s` feels kind of cryptic here.  Unless this is a typo, I would use a full word for the folder.

**Review by brendanshean — APPROVED** [review]
This looks so much better.  Nice work!

File: services/QuillLMS/client/app/bundles/Teacher/components/classrooms/active_classrooms.tsx (hunk 2 above)
Comment (eadams17) [reply services/QuillLMS/client/app/bundles/Teacher/components/classrooms/active_classrooms.tsx]: that's just what we already have as the subfolder for all of our small icons in `quill-cdn`

Comment (emilia-friedberg) [reply services/QuillLMS/client/app/bundles/Teacher/components/classrooms/active_classrooms.tsx]: You can import the `SMALL_ICON_BASE_SRC` from the `Shared` bundle here.

File: services/QuillLMS/client/app/bundles/Teacher/components/classrooms/classroom_student_section.tsx [hunk 3]
```diff
@@ -12,9 +12,8 @@ import ResetStudentPasswordModal from './reset_student_password_modal'
 
 import { DataTable, DropdownInput, Tooltip, helpIcon, warningIcon, } from '../../../Shared/index'
 
-const emptyDeskSrc = `${process.env.CDN_URL}/images/illustrations/empty-desks.svg`
-const lightBulbSrc = `${process.env.CDN_URL}/images/pages/classrooms/lightbulb.svg`
-const questionMarkSrc = `${process.env.CDN_URL}/images/pages/classrooms/question_mark.svg`
+const lightBulbSrc = `${process.env.CDN_URL}/images/icons/s/lightbulb.svg`
+const questionMarkSrc = `${process.env.CDN_URL}/images/icons/s/question-mark-green.svg`
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Teacher/components/classrooms/classroom_student_section.tsx]: Same thing with the base image path here.

File: services/QuillLMS/client/app/bundles/Teacher/components/classrooms/classroom_student_section.tsx [hunk 4]
```diff
@@ -402,17 +401,18 @@ const ClassroomStudentSection = ({
 
   const renderStudentActions = () => {
     if (!classroom.visible) { return null }
+    const noStudentsSelected = selectedStudentIds.length === 0
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Teacher/components/classrooms/classroom_student_section.tsx:404]: Nice cleanup.

File: services/QuillLMS/client/app/bundles/Teacher/components/classrooms/classroom_teacher_section.tsx [hunk 5]
```diff
@@ -266,7 +267,7 @@ export default class ClassroomTeacherSection extends React.Component<ClassroomTe
     if (!classroom.visible) {
       return null
     } else {
-      return <button className="quill-button-archived primary outlined small" onClick={() => this.inviteCoteachers()}>Invite co-teachers</button>
+      return <button className="quill-button primary outlined small focus-on-light" onClick={() => this.inviteCoteachers()}>Invite co-teachers</button>
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Teacher/components/classrooms/classroom_teacher_section.tsx]: I know this isn't part of what you changed, so feel free to ignore, but I think changing this line to `onClick={this.inviteCoteachers}` will have the same function and get rid of an optional linting error.

**Review by emilia-friedberg — APPROVED** [review]
Looks really good! It's nice to see all this clean-up and expanded use of some of our Backpack components. Yay for every replaced `quill-button-archived`!

One tiny thing UI, that I will defer to you and Jack on: having the dividers between the columns in the table makes `Log-in method` and `Activities` look imbalanced to me, because they have more white space on their left side than on their right. If you guys agree, maybe we can mess with some of the other column widths to fix that?

File: services/QuillLMS/client/app/bundles/Teacher/components/classrooms/classroom_teacher_section.tsx (hunk 5 above)
Comment (eadams17) [reply services/QuillLMS/client/app/bundles/Teacher/components/classrooms/classroom_teacher_section.tsx]: will do!

**Pull request merged**